packages/python-sat/test_python_sat.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+from pyodide_test_runner import run_in_pyodide
+
+solvers = [
+    "cadical",
+    "gluecard30",
+    "gluecard41",
+    "glucose30",
+    "glucose41",
+    "maplechrono",
+    "maplecm",
+    "maplesat",
+    "mergesat3",
+    "minicard",
+    "minisat22",
+    "minisat-gh",
+]
+
+
+@pytest.mark.parametrize("name", solvers)
+@run_in_pyodide(packages=["python-sat"])
+def test_solvers(selenium, name):
+
+    from pysat.formula import CNF
+    from pysat.solvers import Solver
+
+    cnf = CNF(from_clauses=[[1, 2, 3], [-1, 2], [-2]])
+
+    with Solver(name=name, bootstrap_with=cnf) as solver:
+        solver.solve()
+        stats = solver.accum_stats()
+        assert "conflicts" in stats, f"No conflicts for {name}"
+        assert "decisions" in stats, f"No decisions for {name}"
```

Comment:
@alexeyignatiev Do you know what is wrong now?
```py
packages/python-sat/test_python_sat.py:33: in test_solvers
    assert "decisions" in stats, f"No decisions for {name}"
E   AssertionError: No decisions for cadical
E   assert 'decisions' in {'restarts': 0, None: 0, 'conflicts': 0}
```
https://app.circleci.com/pipelines/github/hoodmane/pyodide/3607/workflows/3d32cd67-48fd-4736-a51a-7635383a7b33/jobs/42828

Reply:
Honestly, I am not sure. This should not happen. Perhaps it indicates there is an issue in how we handle the stats. It should report something meaningful. For instance, if I run this code in Python on my laptop:
```python
>>> def report_stats():
...     cnf = CNF(from_clauses=[[1, 2, 3], [-1, 2], [-2]])
...
...     for name in solvers:
...         with Solver(name=name, bootstrap_with=cnf) as solver:
...             solver.solve()
...             stats = solver.accum_stats()
...             print(stats, name)
...
>>> from pysat.formula import CNF
>>> from pysat.solvers import Solver
>>> report_stats()
{'restarts': 0, 'conflicts': 0, 'decisions': 0, 'propagations': 3} cadical
{'restarts': 1, 'conflicts': 0, 'decisions': 2, 'propagations': 4} gluecard30
{'restarts': 1, 'conflicts': 0, 'decisions': 2, 'propagations': 4} gluecard41
{'restarts': 1, 'conflicts': 0, 'decisions': 2, 'propagations': 4} glucose30
{'restarts': 1, 'conflicts': 0, 'decisions': 2, 'propagations': 4} glucose41
{'restarts': 1, 'conflicts': 0, 'decisions': 2, 'propagations': 4} maplechrono
{'restarts': 1, 'conflicts': 0, 'decisions': 2, 'propagations': 0} maplecm
{'restarts': 1, 'conflicts': 0, 'decisions': 2, 'propagations': 0} maplesat
{'restarts': 1, 'conflicts': 0, 'decisions': 2, 'propagations': 4} mergesat3
{'restarts': 1, 'conflicts': 0, 'decisions': 2, 'propagations': 4} minicard
{'restarts': 1, 'conflicts': 0, 'decisions': 2, 'propagations': 4} minisat22
{'restarts': 1, 'conflicts': 0, 'decisions': 2, 'propagations': 4} minisat-gh
```
If I run the same code in a Pyolite shell, I get weird stats:
```python
{'restarts': 8397880, None: 0, 'conflicts': 42} cadical
{'restarts': 0, '': 0, 'conflicts': 0, None: 0} gluecard30
{'restarts': 0, '': 0, 'conflicts': 0, None: 0} gluecard41
{'restarts': 0, '': 0, 'conflicts': 0, None: 0} glucose30
{'restarts': 0, '': 0, 'conflicts': 0, None: 0} glucose41
{'restarts': 0, '': 0, 'conflicts': -2, None: 0} maplechrono
{'restarts': 0, '': 0, 'conflicts': -2, None: 0} maplecm
{'restarts': 0, '': 0, 'conflicts': -2, None: 0} maplesat
{'restarts': 0, '': 0, 'conflicts': 0, None: 0} mergesat3
{'restarts': 0, '': 0, 'conflicts': 0, None: 0} minicard
{'restarts': 0, '': 0, 'conflicts': 0, None: 0} minisat22
{'restarts': 0, '': 0, 'conflicts': 0, None: 0} minisat-gh
```
First, as you can see, the numbers are different. Second, even the structure looks different. Empty strings and `None` shouldn't appear as keys in the dictionary. I should say the dictionary is created in C++, i.e. we may be doing something not entirely correctly (although this would be weird because it works just fine in Python).

Reply:
It seems to work? What did you change?

Reply:
Basically, I type-casted the numbers returned by the solvers in their C++ code to Py_ssize_t and took them on the Python side.

Reply:
Yeah looking at the commit it's clear: Python was expecting 4 bytes worth of integer but we gave it 8.